oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/OAuth2AuthorizationRequestRedirectFilterTests.java

```diff
@@ -17,6 +17,10 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.powermock.core.classloader.annotations.PrepareForTest;
```

Comment:
If possible let's avoid powermock

Reply:
Any particular reason to avoid using powermock? There are a few classes in `oauth2-core` that are marked as `final` and so I've used powermock quite a bit already in the oauth2 specific tests.

Reply:
You can mock the interfaces instead. First by mocking the interface, you isolate the test as a unit test which provides a better unit test. Second, powermock is very powerful but historically is not as stable as I'd like. This is due to the fact that it works using internal APIs. Ideally we would get rid of it, but practically it provides a lot of value when working with things like static methods. The alternative is to wrap every static method in an interface declaration which isn't very realistic in my opinion.

Reply:
Gotchya. Yes I can mock `OAuth2AuthorizationRequestRepository`